internal/api/health.go

```diff
@@ -23,18 +32,90 @@ func (h *Health) WithContext(ctx context.Context) {
 	}()
 }
 
+func (h *Health) WithDockerPing(pingFunc func(context.Context) error) {
+	h.dockerPingFunc = pingFunc
+	go h.startDockerHealthCheck(context.Background())
+}
+
+func (h *Health) startDockerHealthCheck(ctx context.Context) {
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			h.checkDockerHealth(ctx)
+		}
+	}
+}
+
+func (h *Health) checkDockerHealth(ctx context.Context) {
+	if h.dockerPingFunc == nil {
+		return
+	}
+
+	pingCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	err := h.dockerPingFunc(pingCtx)
+
+	h.dockerPingMutex.Lock()
+	defer h.dockerPingMutex.Unlock()
+
+	h.dockerPingLastCheck = time.Now()
+	if err != nil {
+		h.dockerPingStatus = false
+		h.dockerPingMessage = "Docker API connection issue: " + err.Error()
+	} else {
+		h.dockerPingStatus = true
+		h.dockerPingMessage = "Docker API connection healthy"
+	}
+}
+
 func (h *Health) ServeHTTP(c *gin.Context) {
 	statusCode := http.StatusOK
 	if h.terminating {
 		statusCode = h.TerminatingStatusCode
+		c.String(statusCode, http.StatusText(statusCode))
+		return
+	}
+
+	// Check Docker health status if we're using Docker provider
+	h.dockerPingMutex.RLock()
+	dockerHealthy := h.dockerPingStatus
+	dockerMessage := h.dockerPingMessage
+	lastCheck := h.dockerPingLastCheck
+	h.dockerPingMutex.RUnlock()
+
+	// If Docker is not healthy and Docker provider is being used, return unhealthy
+	if h.dockerPingFunc != nil && !dockerHealthy {
```

Comment:
Call the `providerHealthFunc` and if error is not nil, write the error message on the body.

I would remove the headers you added completely